src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate2Enumerator.cs

```diff
@@ -2,30 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Diagnostics;
 using System.Collections;
-using System.Collections.Generic;
 
 namespace System.Security.Cryptography.X509Certificates
 {
     public sealed class X509Certificate2Enumerator : IEnumerator
     {
-        // This is a mutable struct enumerator, so don't mark it as readonly.
-        private List<X509Certificate>.Enumerator _enumerator;
+        private IEnumerator _enumerator;
 
         internal X509Certificate2Enumerator(X509Certificate2Collection collection)
         {
-            Debug.Assert(collection != null);
+            if (collection == null)
+                throw new ArgumentNullException(nameof(collection));
```

Comment:
The value of Debug.Assert(x != null) is finding bugs upstream (eg at the point the value is stored) .. it doesn't give me anything if it's getting dereferenced in the next line of code. I'll remove the check